test/module/shared_model/mock_objects_factories/mock_query_factory.cpp

```diff
@@ -64,12 +64,17 @@ MockQueryFactory::FactoryResult<MockGetAccountDetail>
 MockQueryFactory::constructGetAccountDetail(
     const types::AccountIdType &account_id,
     boost::optional<types::AccountDetailKeyType> key,
-    boost::optional<types::AccountIdType> writer) const {
+    boost::optional<types::AccountIdType> writer,
+    boost::optional<const AccountDetailPaginationMeta &> pagination_meta)
+    const {
   return createFactoryResult<MockGetAccountDetail>(
-      [&account_id, &key, &writer](MockGetAccountDetail &mock) {
+      [&account_id, &key, &writer, &pagination_meta](
+          MockGetAccountDetail &mock) {
         EXPECT_CALL(mock, accountId()).WillRepeatedly(ReturnRef(account_id));
```

Comment:
Not an issue for this particular pull request, but `ReturnRef` should be replaced with `ReturnRefOfCopy` to simplify the lifetime management.